File: work_1/content/作业一.py
import random
import struct

# 生成5个32位无符号整数
random_numbers = [random.randint(0, 2**32 - 1) for _ in range(5)]
print("生成的随机数:", random_numbers)

# 写入input.txt
with open("input.txt", "w") as f:
    f.write(' '.join(map(str, random_numbers)))

# 从input.txt读取
with open("input.txt", "r") as f:
    content = f.read().strip()

# 转换为数字列表
my_list = list(map(int, content.split()))

# 计算平均数（float32）
average = sum(my_list) / len(my_list)

# 写入output.txt（二进制格式）
with open("output.txt", "wb") as f:
    f.write(struct.pack('!f', average))

print(f"这些数的平均数为：{average}")
print("结果已经写入output.txt")

File: work_1/content/作业一.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <random>
#include <cstdint>
using namespace std;
int main()
{
    // 初始化随机数生成器
    random_device rd;
    mt19937 gen(rd());
    uniform_int_distribution<uint32_t> dis(0, UINT32_MAX);
    // 生成5个32位无符号整数
    vector<uint32_t> arr;
    for(int i = 0; i < 5; i++) {
        uint32_t num = dis(gen);
        arr.push_back(num);
        cout << num << " ";
    }
    cout << endl;
    // 写入input.txt
    ofstream output("input.txt");
    if(!output) {
        cout << "无法打开文件" << endl;
        return 0;
    }
    for(uint32_t n : arr) {
        output << n << " ";
    }
    output.close();
    cout << "已保存以上数字到input.txt" << endl;

    // 从input.txt读取
    ifstream input("input.txt");
    if(!input) {
        cout << "无法打开文件" << endl;
        return 0;
    }
    vector<uint32_t> arr2;
    uint32_t x;
    while(input >> x) {
        arr2.push_back(x);
    }
    input.close();

    // 计算平均数
    uint64_t sum = 0;
    for(uint32_t n : arr2) {
        sum += n;
    }
    float average = static_cast<float>(sum) / arr2.size();

    // 写入output.txt（二进制格式）
    ofstream output2("output.txt", ios::binary);
    if(!output2) {
        cout << "无法打开文件" << endl;
        return 0;
    }
    output2.write(reinterpret_cast<const char*>(&average), sizeof(float));
    output2.close();

    cout << "这些数的平均数为：" << average << endl;
    cout << "结果已经写入output.txt（二进制格式）" << endl;

    return 0;
}